### DISCUSSION TITLE
The command "node "C:\wamp64\www\Masampo-P1\vendor\spatie\browsershot\src/../bin/browser.cjs" "{""url"":""file:\/\/C:\\Users\\Kahilu\\AppData\\Local\\Temp\\1571387470-0974524001712322563\\index.html"",""action"":""pdf"",""options"":{""path"":""welcome.pdf"",""args"":[],""viewport"":{""width"":800,""height"":600},""displayHeaderFooter"":false,""printBackground"":true}}"" failed. Exit Code: 1(General error) Working directory: C:\wamp64\www\Masampo-P1\public Output: ================ Error Output: ================ 'node' is not recognized as an internal or external command, operable program or batch file.[Bug]:

### DISCUSSION THREAD

**Discussion by KahiluChipango** [spatie/laravel-pdf#119] [Q&A]
### What happened?

i expect it to work perfectly.

### How to reproduce the bug

just run Php artisan serve after installing Node and npm install puppeteer and also installing composer require spatie/laravel-pdf

### Package Version

1.4

### PHP Version

8.2.0

### Laravel Version

11.0.0

### Which operating systems does with happen with?

Windows

### Notes

This package was created on MacOs and is not properly implemented to look for Dependencies like node if its installed

**Comment by Professor93M** [comment]
The same happened with me!!!

**Comment by Professor93M** [comment]
I fix this issue.
You can try this:

- Remove **vendor** folder and run `composer install`
- try to run `composer update`

and it's will work with you

**Comment by hamannjames** [comment]
Having this same problem on a mac and the above did not fix it. What can I do? There is nothing weird about my system setup. I can run `npm root -g` and get my node path, using nvm, but browsershot cannot find it

**Comment by hamannjames** [comment]
Alright, for anyone wondering how to get around this without modifying vendor files, I had to find my local npm and node binaries and call

```
pdf()->withBrowsershot(function (Browsershot $browsershot) {
    $browsershot->setNodeBinary('/path/to/node');
    $browsershot->setNpmBinary('/path/to/npm');
})
```

This let me use the package functionality. I think this might be an issue with using nvm locally? Regardless, for a staging and prod solution you would probably need to set those as env variables

**Comment by mateusjunges** [comment]
Anyone got it working with nvm?

**Reply by hamannjames** [reply]
I had to find my local node and npm binaries using `which node` and then set them inside the `pdf()->withBrowsershot()` callback. I set them as env files so that, if necessary, I could also set them for staging and production.

**Reply by ThantHtetAung101** [reply]
Where to put pdf()->withBrowsershot()? 
This is my code
```
$pdf = Pdf::view('pos.kitchen-receipt', ['cart' => $cart, 'table' => $table]);
        return $pdf->save('testing.pdf');
```

**Reply by hamannjames** [reply]
You can do it after view. I think you just need to do it before save

https://spatie.be/docs/laravel-pdf/v1/advanced-usage/customizing-browsershot

**Reply by ThantHtetAung101** [reply]
Node problem is solved but now puppeteer is giving me headaches. I installed it globally and locally in project too. Didn't work. 
```
[Error: ENOENT: no such file or directory, mkdtemp 'undefined\temp\puppeteer_dev_chrome_profile-XXXXXX'] {
errno: -4058,
code: 'ENOENT',
syscall: 'mkdtemp',
path: 'undefined\temp\puppeteer_dev_chrome_profile-XXXXXX'
}
```
I also tried adding chrome execution path but no luck

**Reply by tsitsi98** [reply]
@ThantHtetAung101 have you found any solution to this problem with the puppeteer?

**Reply by ThantHtetAung101** [reply]
@tsitsi98 This worked for me.
https://github.com/spatie/browsershot/discussions/728#discussioncomment-8355492

**Reply by tsitsi98** [reply]
@ThantHtetAung101 hey thanks now i can generate, the only problem now is that the tailwind im using is not working and i cant style it! i cant include the @ vite where i have all my resources

**Comment by mateusjunges** [comment]
Update: 

I created a symlink for `/Users/mateusjunges/.nvm/versions/node/v22.1.0/bin/node` to `/opt/homebrew/bin/node`, and `/Users/mateusjunges/.nvm/versions/node/v22.1.0/npm` to `/opt/homebrew/bin/npm`. It does not thrown an exception, but I still can not view the pdf on my browser. This is the error now:

![image](https://github.com/spatie/laravel-pdf/assets/19756164/9f7d07c1-914a-4c19-899a-0805e56bff5f)

**Comment by parmar-sagar** [comment]
Using **->noSandbox()** ensures that Browsershot operates without sandboxing restrictions, which resolves common issues related to sandboxing errors when generating PDFs. This approach should enable successful PDF generation in your Laravel application, particularly addressing scenarios where sandboxing settings interfere with Browsershot's functionality.

```php
use Spatie\LaravelPdf\Facades\Pdf;
use Spatie\Browsershot\Browsershot;

// Example data (replace with your actual data)
$datas = [];

// Generate PDF with Browsershot, addressing "node" command issue
PDF::view('path.to.view', $datas)
    ->format('A4')
    ->margins(10, 10, 10, 10)
    ->withBrowsershot(function (Browsershot $browsershot) {
        // Ensure Browsershot works without node command recognition issues
        $browsershot->noNodeInstallation()->noSandbox()->scale(0.5);
    })
    ->save('invoice.pdf');

``` 

**Note**: This solution is particularly effective on Windows where sandboxing configurations may differ from Unix-based systems.

**Comment by elvispdosreis** [comment]
I even managed to run it on Windows, but with simple html `<strong>test</strong>` and it was extremely slow
